src/ReaderScreen/utils/gutkaScript.js

```diff
@@ -116,45 +45,124 @@ function scrollFunc(e) {
   }
   scrollFunc.y = window.pageYOffset;
 }
-window.onscroll = scrollFunc;
 
-const handleTouchEnd=()=>{
+const getScrollPercent=()=> {
+  return window.pageYOffset / (document.body.scrollHeight - window.innerHeight);
+}
+
+const fadeInEffect = () => {
+  let fadeTarget = ${body};
+  fadeTarget.style.opacity = 0;
+  let fadeVal = 0;
+  let fadeEffect = setInterval(() => {
+    if (fadeVal < 1) {
+      fadeVal = Number(fadeVal) + 0.1;
+      fadeTarget.style.opacity = fadeVal;
+    } else {
+      fadeTarget.style.opacity = 1;
+      clearInterval(fadeEffect);
+    }
+  }, 100);
+};
```

Comment:
@amitojsingh How about if we use `requestAnimationFrame` here? I looked briefly and it's said that the animation would be more smoother and efficient as well.

```javascript
const fadeInEffect = () => {
  let fadeTarget = ${body}
  fadeTarget.style.opacity = 0;
  
  let fadeVal = 0;
  const fadeSpeed = 0.1; // Adjust for a smoother effect

  const fade = () => {
    fadeVal += fadeSpeed;
    if (fadeVal >= 1) {
      fadeTarget.style.opacity = 1; // Ensure it ends at exactly 1
      return;
    }
    fadeTarget.style.opacity = fadeVal;
    requestAnimationFrame(fade);
  };

  requestAnimationFrame(fade);
};
```

Reply:
I tried the above code Veer ji. Here's the fade effect after the code. 

https://github.com/user-attachments/assets/d78c7816-e3af-4e8d-8e7e-6f67392a39f5

And this was before we updated the code. 

https://github.com/user-attachments/assets/aa674248-1c92-44be-8d28-8b46b568a13b

Ps: I feel like the previous fade effect is better.